hack/provision.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# DESCRIPTION:
+# This is a script that bootstraps basic content in the OpenShift server
+# to facilitate development flows and users that want to kick the tires.
+# 
+# PROVISIONS:
+# * Users
+# ** test-cluster-admin
+# ** test-admin
+# * Docker registry
+# * OpenShift router
+# * OpenShift templates
+
+# INSTRUCTIONS:
+# $ git clone ...
+# $ make all
+# $ openshift start
+# $ hack/provision.sh
+
+# Export necessary environment variables for resources created on openshift start
+export ADMIN_KUBECONFIG=`pwd`/openshift.local.config/master/admin.kubeconfig
+export REGISTRY_KUBECONFIG=`pwd`/openshift.local.config/master/openshift-registry.kubeconfig
+export ROUTER_KUBECONFIG=`pwd`/openshift.local.config/master/openshift-router.kubeconfig
+export CA=`pwd`/openshift.local.config/master
+
+# Modify permissions on each config
+sudo chmod -R a+rwx "$CA"
+sudo chmod a+rwX "$ADMIN_KUBECONFIG"
+sudo chmod a+rwX "$REGISTRY_KUBECONFIG"
+sudo chmod a+rwX "$ROUTER_KUBECONFIG"
+
+# Run as admin
+export KUBECONFIG=$ADMIN_KUBECONFIG
+
+# Configure test users
+oadm policy add-cluster-role-to-user cluster-admin test-cluster-admin
+oadm policy add-role-to-user view test-admin
+
+# Install a docker registry on the cluster
+oadm registry --create --credentials="$REGISTRY_KUBECONFIG" --latest-images
+
+# Install an OpenShift router
+echo '{"kind":"ServiceAccount","apiVersion":"v1","metadata":{"name":"router"}}' | oc create -n default -f -
+oc get scc privileged -o yaml > /tmp/priv.yaml
+echo "- system:serviceaccount:default:router" >> /tmp/priv.yaml
+oc replace -f /tmp/priv.yaml
+pushd /tmp
+oadm ca create-server-cert --signer-cert=${CA}/ca.crt \
+  --signer-key=${CA}/ca.key --signer-serial=${CA}/ca.serial.txt \
+  --hostnames='*.router.default.svc.cluster.local' \
+  --cert=router.crt --key=router.key
+cat router.crt router.key ${CA}/ca.crt > router.pem
+oadm router --default-cert=router.pem --credentials="${ROUTER_KUBECONFIG}" --service-account=router
+popd
+
+# Install templates from origin
+oc create -n openshift -f ./examples/db-templates
+oc create -n openshift -f ./examples/image-streams/image-streams-centos7.json
```

Comment:
this one isn't a template, it's the imagestreams... break it out separately and also show the rhel7 equivalent, commented out.  Or make this script take an arg which controls whether you get RHEL or CENTOS imagestreams.  (the imagestream names are the same, so you can't install both, at least not both to the openshift namespace)